### ISSUE TITLE
JS, EditorConfig, and evil-shift-width

### ISSUE THREAD

**Issue by jupl** [syl20bnr/spacemacs#5810]
#### Description
When opening a JS file, `evil-shift-width` is always set to 4 and ignores Editorconfig. This does not appear to be happening with other file types affected by Editorconfig. Also this appears to have started to happen recently.

#### Reproduction guide
- Start Emacs
- Ensure `editorconfig` is installed and `editorconfig-mode` is enabled
- Open a JS file such that it is covered by an `.editorconfig` where `index_size` is not 4
- `M-:` and eval `evil-shift-width`
- Open a non JS file such that it is covered by an `.editorconfig` where `index_size` is not 4
- `M-:` and eval `evil-shift-width`

*Observed behaviour:*
`evil-shift-width` for the JS file has the value `4`

*Expected behaviour:*
`evil-shift-width` should not have the value `4` for both files

#### System Info
- OS: darwin
- Emacs: 24.5.1
- Spacemacs: 0.105.18
- Spacemacs branch: develop (rev. 5ee970f)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
((clojure :variables clojure-enable-fancify-symbols t)
 colors dockerfile emacs-lisp html github markdown osx react restclient shell shell-scripts syntax-checking vimscript yaml
 (auto-completion :variables auto-completion-enable-help-tooltip t)
 (git :variables git-magit-status-fullscreen t magit-push-always-verify nil)
 (javascript :variables js2-mode-show-parse-errors nil js2-mode-show-strict-warnings nil)
 (org :variables org-src-preserve-indentation t org-startup-folded 'showall)
 (version-control :variables version-control-diff-tool 'diff-hl))
```

**Comment by TheBB** [comment]
`evil-shift-width` should be equal to

* `js-indent-level` in `js-mode`
* `js2-basic-offset` in `js2-mode`
* `js3-indent-level` in `js3-mode`

Can you confirm that this is or is not the case given the mode you are using?

**Comment by jupl** [comment]
It is not. `js2-basic-offset` (`js2-mode` from the `js` layer) is set to `2` whereas `evil-shift-width` is at 4.

**Comment by TheBB** [comment]
Okay, I found the problem. `js2-mode` is derived from `js-mode` so it triggers the `js-mode` entry since it comes first.

**Comment by orther** [comment]
Do we know if this fix will make it into official release soon?

_Please excuse my ignorance if there is a standard process or time period required before a fix moves from `develop` to `master` that I am unaware of._

**Comment by StreakyCobra** [comment]
No, there isn't any rule about the release process yet, it's when @syl20bnr think it's ready. I'm not seeing a release really soon because he has still some changes he want to do before the next one. But who knows… @syl20bnr thought are impenetrable :smile: Anyway, usually the `develop` branch is stable enough for non-critical use, so you may use it in the meantime.

**Comment by orther** [comment]
Thanks for the response @StreakyCobra 

I switched to using the develop branch last week and aside from a few minor adjustments it seems to be working well. I guess that's how I will proceed.

**Comment by syl20bnr** [comment]
The release will happen in June for sure. There is more that 1k commits to release, this project is crazy :-D

**Comment by eyko** [comment]
I'm running on `develop`, however my `evil-shift-width` is correctly set to 2, however both my `js-indent-level` and `js2-basic-offset` are set to 4. Manually hitting tab jumps at 2 space intervals but hitting enter indents at 4 spaces.

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
